app/Espo/Listeners/Metadata.php

```diff
@@ -445,6 +447,29 @@ protected function setAssetThumbSize(array $data): array
         return $data;
     }
 
+
+    public function addArchive(array $data){
+        foreach ($data['scopes'] as $scope => $row) {
+            if (!empty($row['hasArchive'])) {
+                if (!isset($data['entityDefs'][$scope]['fields']['isArchived']['type'])) {
+                    $data['entityDefs'][$scope]['fields']['isArchived']['type'] = 'bool';
+                }
+
+                if (!isset($data['entityDefs'][$scope]['fields']['isArchived']['notNull'])) {
+                    $data['entityDefs'][$scope]['fields']['isArchived']['notNull'] = true;
+                }
+            }
+        }
+
+        foreach ($data['entityDefs'] as $entity => $row) {
+            if (isset($row['fields']['isArchived']['type']) && $row['fields']['isArchived']['type'] == 'bool') {
+                $data['clientDefs'][$entity]['filterList'][] = 'withArchived';
```

Comment:
Why 'filterList'? It should be boolFilterList

Reply:
Alex changed the requirements, now the filter is shown in the primary filter list